spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpMessageMapper.java

```diff
@@ -208,6 +228,9 @@ protected final void addCustomHeaders(TcpConnection connection,
 
 	@Override
 	public Object fromMessage(Message<?> message) throws Exception {
+		if (this.bytesMessageMapper != null) {
+			return this.bytesMessageMapper.fromMessage(message);
+		}
 		if (this.stringToBytes) {
```

Comment:
We have to document somewhere that `stringToBytes` is ignored if `bytesMessageMapper` is provided. Or reconsider logic and don't use `bytesMessageMapper ` unconditionally...